internal/dcache/rpc/server/handler.go

```diff
@@ -670,9 +767,75 @@ func (mv *mvInfo) refreshFromClustermap() error {
+		// It marks reservedSpace in the mvInfo and rvInfo to reserve the space needed for sync'ing the MV
+		// replica. Normally this reserved space would be deducted from rvInfo.reservedSpace as part of
+		// the EndSync processing, after the sync has copied data to the new MV replica. At this point
+		// mvInfo.totalChunkBytes will be increased by mvInfo.reservedSpace, and rvInfo.reservedSpace will
+		// be reduced by mvInfo.reservedSpace and mvInfo.reservedSpace will be set to 0.
+		// If the mvInfo has the state as StateOutOfSync while clustermap has the state as StateOffline,
+		// this means the fix-mv workflow didn't complete successfully so we need to rollback the reserved
+		// space changes.
+		//
+		if oldRvNameAndState.State == string(dcache.StateOutOfSync) && rvState == dcache.StateOffline {
+			common.Assert(mv.rv.reservedSpace.Load() >= mv.reservedSpace.Load(),
+				mv.rv.reservedSpace.Load(), mv.reservedSpace.Load())
+			mv.rv.reservedSpace.Add(-mv.reservedSpace.Load())
+			mv.reservedSpace.Store(0)
+		}
+
+		//
+		// StateSyncing -> StateOutOfSync
+		//
+		// Consider the following case:
+		// client is running the sync-mv workflow and decides to sync rv0/mv0 -> rv2/mv0
+		// it'll send a StartSync request and the mvInfo.syncJobs will have a new sync job added.
+		// If all went well, the client would send EndSync on completion of the sync job, which will
+		// remove the sync job from mvInfo.syncJobs, but let's say sync didn't proceed normally and
+		// was aborted. Next time when the mv was again picked for sync'ing, this time rv0 went offline
+		// and hence rv1 was picked as the source replica, so now the client sends a fresh StartSync
+		// request for rv1/mv0 -> rv2/mv0. This will find the rvInfo in StateSyncing which it doesn't
+		// expect so refreshFromClustermap() is called. If we do not remove the older syncJob from
+		// mvInfo.syncJobs, we will have multiple syncJobs queued for a target RV. Note that we consider
+		// an mvInfo with more than one syncJobs as being a source replica (ref mvInfo.isSourceOrTargetOfSync).
+		// So we wrongly treat it as a being a source replica.
+		// Hence whenever we have refreshFromClustermap() see a state transition from StateSyncing to
+		// StateOutOfSync, it means that it's this case and we must clear the old syncJob.
+		//
+		if oldRvNameAndState.State == string(dcache.StateSyncing) && rvState == dcache.StateOutOfSync {
+			//
+			// Only a target replica can be in StateSyncing and a target replica MUST have one and
+			// only one syncJob, clear that.
+			//
+			common.Assert(mv.syncJobsCount.Load() == 1, rvName, mv.mvName, mv.syncJobsCount.Load(),
+				mv.getSyncJobs(), rpc.ComponentRVsToString(mv.componentRVs))
+
+			log.Warn("mvInfo::refreshFromClustermap: %s/%s, clearing old syncJob left from previous incomplete sync attempt, syncJobs: %+v",
+				rvName, mv.mvName, mv.getSyncJobs())
+
+			mv.deleteAllSyncJobs()
+		}
```

Comment:
We should also decrement the rvInfo.reservedSpace by mvInfo.reservedSpace and set mvInfo.reservedSpace to 0.

Reply:
since the space is reserved in JoinMV when we move an RV to StateOutOfSync, and we are still in StateOutOfSync, we should not decrease the space.